README.md

```diff
@@ -283,22 +280,117 @@ The post-split method should be written in `steps/split.py` and should accept th
 It should return a triple representing the processed train, validation and test datasets. `steps/split.py` contains an example placeholder function.
 
 The split step is configured by the `steps.split` section in `recipe.yaml` as follows:
+
 <details>
-<summary><strong><u>Full configuration reference</u></strong></summary>
+<summary><strong><u>Using: "custom"</u></strong></summary>
+
+- `split_method`: string. Required. The user-defined split function should be written in steps/split.py, 
+and should return a Series with each element to be either 'TRAINING', 'VALIDATION' or 'TEST' indicating
+whether each  row should should be part of which split. 
+<u>Example</u>:
+  ```
+  split_method: custom_split
+  ```
+
+- `post_split_filter_method`: string. Optional.
+Name of the method specified in `steps/split.py` used to "post-process" the split datasets.  
+This procedure is meant for removing/filtering records, or other cleaning processes. Arbitrary data transformations 
+should be done in the transform step.
+<u>Example</u>:
+  ```
+  post_split_filter_method: create_dataset_filter
+  ```
+
+  Example config in [`recipe.yaml`](https://github.com/mlflow/recipes-classification-template/blob/main/recipe.yaml):
+  ```
+  steps:
+    split:
+      using: custom
+      split_method: custom_split
+      post_split_filter_method: create_dataset_filter
+  ```
+  Example create_dataset_filter function for [`split.py`](https://github.com/mlflow/recipes-classification-template/blob/main/steps/split.py)
+  ```python
+  def custom_split(df):
+    """
+    Mark rows of the ingested datasets to be split into training, validation, and test datasets.
+    :param dataset: The dataset produced by the data ingestion step.
+    :return: Series of strings with each element to be either 'TRAINING', 'VALIDATION' or 'TEST'
+    """
+    from pandas import Series
+
+    splits = Series("TRAINING", index=range(len(df)))
+    # 3rd quarter is validation data
+    splits[df["year"] >= 3] = "VALIDATION"
+    # 4th quarter is testing data
+    splits[df["year"] >= 4] = "TEST"
+
+    return splits
+
+  def create_dataset_filter(dataset: DataFrame) -> Series(bool):
+    """
+    Mark rows of the split datasets to be additionally filtered. This function will be called on
+    the training, validation, and test datasets.
+    :param dataset: The {train,validation,test} dataset produced by the data splitting procedure.
+    :return: A Series indicating whether each row should be filtered
+    """
+
+    return (
+        (dataset["fare_amount"] > 0)
+        & (dataset["trip_distance"] < 400)
+        & (dataset["trip_distance"] > 0)
+        & (dataset["fare_amount"] < 1000)
+    ) | (~dataset.isna().any(axis=1))
+  ```
 
-- `split_ratios`: list. Optional.  
-A YAML list specifying the ratios by which to split the dataset into training, validation and test sets.  
+
+</details>
+
+<details>
+<summary><strong><u>Using: "split_ratio"</u></strong></summary>
```

Comment:
`split_ratios` instead of `split_ratio` to be consistent with the parameter required? This may need to change the code as well.